### PR TITLE
refactor(block): mode variant classes consistent with other components

### PR DESCRIPTION
**Pull request by EmelieLitwin**
**Describe pull-request**  
Changing mode variant classes to be consistent with other components

**Solving issue**  
Fixes: [AB#3187](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/3187)

**Suggested test steps**
- [ ] Browser testing (Chrome, Safari, Firefox) 
- [ ] Keyboard operability
- [ ] Interactive elements have labels.
- [ ] Storybook controls
- [ ] Design/controls/props is aligned with other components 
- [ ] Dark/light mode and variants 
- [ ] Input fields – values should be displayed properly 
- [ ] Events

**Screenshots**  
_If applicable, add screenshots to help explain_

**Additional context**  
_Add any other context about the pull-request here._

### CODE REVIEW

File: tegel/src/components/block/block-theme-vars.scss [hunk 1]
```diff
@@ -1,18 +1,42 @@
...
+    --sdds-block-background: var(--sdds-block-background-primary);
+    --sdds-block-background-nested: var(--sdds-block-background-nested-primary);
   }
+  .sdds-mode-variant-secondary {
+    --sdds-block-background: var(--sdds-block-background-secondary);
+    --sdds-block-background-nested: var(--sdds-block-background-nested-secondary);
+  }
+}
 
+:root,
```
Comment (mJarsater) [inline tegel/src/components/block/block-theme-vars.scss:23]: Does this :root have to be removed?

Comment (mJarsater) [reply tegel/src/components/block/block-theme-vars.scss:23]: I see it still works 🤔 Anyhow, nice job!

**Comment by mJarsater** [comment]
I guess we should align this one with: https://github.com/scania-digital-design-system/sdds/pull/800

File: tegel/src/components/block/block-theme-vars.scss (hunk 1 above)
Comment (EmelieLitwin) [reply tegel/src/components/block/block-theme-vars.scss:23]: Good catch, I missed it! I'm not sure what happens, but it doesn't work after merging to main

File: tegel/src/components/block/sdds-block.scss [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+}
+.sdds-mode-variant-primary {
+  --sdds-block-background: var(--sdds-block-background-primary);
+  --sdds-block-background-nested: var(--sdds-block-background-nested-primary);
+}
+.sdds-mode-variant-secondary {
+  --sdds-block-background: var(--sdds-block-background-secondary);
+  --sdds-block-background-nested: var(--sdds-block-background-nested-secondary);
+}
```
Comment (mJarsater) [inline tegel/src/components/block/sdds-block.scss:21]:
```suggestion
}

```

**Comment by mJarsater** [comment]
All the block stuff look good to me. Im guessing the textarea/textfield/icon stuff shouldn't really be part of the PR. But maybe they got in there by accident?

File: tegel/src/components/block/block-theme-vars.scss [hunk 3]
```diff
@@ -1,18 +1,41 @@
...
+  }
+  .sdds-mode-variant-secondary {
+    --sdds-block-background: var(--sdds-block-background-secondary);
+    --sdds-block-background-nested: var(--sdds-block-background-nested-secondary);
   }
+}
 
+.sdds-theme-dark {
+  --sdds-block-color: var(--sdds-grey-50);
+  --sdds-block-background-primary: var(--sdds-grey-900);
```
Comment (nathalielindqvist) [inline tegel/src/components/block/block-theme-vars.scss:25]: The color when switching between mode variant don't match between the native component and the web component. In the web component the base background color and the nested one simply switch position between the primary and secondary variant. In the native a whole new color (Grey846 I think) is introduced. Do we know which one is correct? Or should there be a difference between how the native and the web component are behaving here?
image
image
image
image

**Review by mJarsater — APPROVED** [review]
LGTM 🦄

**Comment by charlottelorensson** [comment]
Visually I would suggest to perhaps remove one level of the nested blocks so that it aligns with the documentation on the site. I'd also shorten the length of the text in the block so that it appears easier to read even though it is a placeholder :)

**Comment by EmelieLitwin** [comment]
> Visually I would suggest to perhaps remove one level of the nested blocks so that it aligns with the documentation on the site. I'd also shorten the length of the text in the block so that it appears easier to read even though it is a placeholder :)

GREAT feedback Charlotte! Thank you!

**Pull request merged**